AsyncSwift/Observed/EventView+Observed.swift

```diff
@@ -6,9 +6,70 @@
 //
 
 import Foundation
+import SwiftUI
 
 extension EventView {
     final class Observed: ObservableObject {
-        
+
+        @Published var response = JSONResponse()
+        @Published var eventStatus: EventStatus = .upcoming
+
+        init() {
+            fetchJson()
+        }
+
+        func fetchJson() {
+            guard let url = URL(string: "https://async-swift.github.io/jsonstorage/asyncswift.json") else { return }
+            let request = URLRequest(url: url)
+            let dataTask = URLSession.shared.dataTask(with: request) { data, response, error in
+                if let error = error {
+                    print(error.localizedDescription)
+                    return
+                }
+                guard let response = response as? HTTPURLResponse else { return }
+                if response.statusCode == 200 {
+                    guard let data = data else { return }
+                    DispatchQueue.main.async { [weak self] in
+                        if let self = self {
+                            do {
+                                let decodedData = try JSONDecoder().decode(JSONResponse.self, from: data)
+                                withAnimation {
+                                    self.response = decodedData
+                                }
+                                self.calculateEventStatus()
+                            } catch let error {
+                                print("❌ \(error.localizedDescription)")
+                            }
+                        }
+                    }
+                }
+            }
+            dataTask.resume()
+        }
+
+        func calculateEventStatus() {
+            let formatter = DateFormatter.calendarFormatter
+            let start = formatter.date(from: self.response.event.startDate)
+            let end = formatter.date(from: self.response.event.endDate)
+            let current = Date()
+
+            if let start = start, let end = end {
```

Comment:
```suggestion
            guard
                let start = formatter.date(from: event.startDate),
                let end = formatter.date(from: event.endDate)
            else { return }

            let current = Date()
```

guard let을 사용한다면 property 정의와 Optional 확인을 한 번에 할 수 있습니다

Reply:
맞네요 바로 수정하겠습니다